From dcaca3ac17a99ba5d2b84f57a90125d00a95a64f Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 3 Apr 2019 12:02:39 +0200
Subject: [PATCH] ltq_spinand: Fix compile warning

The deadline variable is only used conditionally based on the compile flag.
---
 drivers/mtd/ltq-spinand/ltq_spinand.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/mtd/ltq-spinand/ltq_spinand.c b/drivers/mtd/ltq-spinand/ltq_spinand.c
index abaa3fbbf79e..77f955078cc9 100644
--- a/drivers/mtd/ltq-spinand/ltq_spinand.c
+++ b/drivers/mtd/ltq-spinand/ltq_spinand.c
@@ -514,11 +514,13 @@ static int spinand_read_status(struct spi_device *spi, uint8_t *status)
 #define MAX_WAIT_MS  40
 static int wait_till_ready(struct spi_device *spi)
 {
-	unsigned long deadline;
 	int retval;
 	u8 stat = 0;
+#ifndef NOTIMEOUT
+	unsigned long deadline;
+#endif
 
-	#ifdef NOTIMEOUT
+#ifdef NOTIMEOUT
 
 	do {
 		retval = spinand_read_status(spi, &stat);
@@ -530,8 +532,7 @@ static int wait_till_ready(struct spi_device *spi)
 		cond_resched();
 	} while (1);
 
-
-	#else /* NOTIMEOUT */
+#else /* NOTIMEOUT */
 
 	deadline = jiffies + msecs_to_jiffies(MAX_WAIT_MS);
 	do {
@@ -544,7 +545,7 @@ static int wait_till_ready(struct spi_device *spi)
 		cond_resched();
 	} while (!time_after_eq(jiffies, deadline));
 
-	#endif /* NOTIMEOUT */
+#endif /* NOTIMEOUT */
 
 	if ((stat & 0x1) == 0)
 		return 0;
